### ISSUE TITLE
Make profiling alert message soft (and configurable)

### ISSUE THREAD

**Issue by lastquestion** [lastquestion/explain-pause-mode#13]
Currently the profiling alert message prints out to minibuffer no matter what is going on. If profiling happens this can interrupt the user while they are using the minibuffer for input. I had originally thought this was a good idea - because profiling should be relatively rare.

However, if we want people to use this thing nonstop, continuously, in the background, it needs to be really unobtrusive. 

Make the profiling messages just like the alert messages. The "alert messages" already implement "soft alerting" where it only prints stuff if the minibuffer isn't in use, do the same thing.

**Comment by wasamasa** [comment]
Is there a reason why both can't be disabled by the same customizable?

**Comment by lastquestion** [comment]
> Is there a reason why both can't be disabled by the same customizable?

No reason. At first, I liked the idea that "some messages" are more important than others. But on second thought, you have a point. The profiles and the "alert messages" are basically the same, they are obtrusively interrupting the user.

Now I'm starting to think: really messages shouldn't appear in "normal mode", and occasionally the thing should just say "Hey, emacs was slow N times in the last hour. Check out the report <here>". 

So basically, maybe one main customizable: "developer mode" = like devtools, you get "polite" echo area alerts for both; or "normal mode" = only summary messages every now and again.

**Comment by wasamasa** [comment]
Maybe even a tristate:

```elisp
(defcustom explain-pause-alert-style 'normal
  "How often should explain-pause alert you about long pauses in the mini-buffer?"
  :type '(choice (const :tag "Always" developer)
                 (const :tag "Every now and then" normal)
                 (const :tag "Never" silent)))
```

**Comment by lastquestion** [comment]
❤️ 😻 💯 

That's so great, thank you for prodding me to remember I'm in emacs and should do things the emacs way. I'll use this entire thing. super appreciate 👍

**Comment by wasamasa** [comment]
Oh and you probably want to define a group and add all customizables to it for `M-x customize-group` to work.

**Comment by lastquestion** [comment]
I think I did the customize-group stuff correctly, but I will be honest I'm literally just following the documentation and grepping what the emacs built-in .el files do, I really am not a elisp developer by any stretch of the imagination... I have this
```
(defgroup explain-pause nil
  "Explain pauses in Emacs"
  :prefix "explain-pause-")
```

https://github.com/lastquestion/explain-pause-mode/blob/master/explain-pause-mode.el#L46-L48

which should work, right?

**Comment by wasamasa** [comment]
Yes, but then you need to use `:group explain-pause` in every customizable to link them up.

**Comment by lastquestion** [comment]
So I asked because it seems to work without it :-)

I investigated and discovered this commit https://github.com/emacs-mirror/emacs/commit/805204f3efdf3e9cb3ccd2783b7ba44f8c1fdb5a in 2016 - which added the text in the docs that I read. The bug report: https://lists.gnu.org/archive/html/bug-gnu-emacs/2016-05/msg00132.html

which seems to be ambivalent on whether it is a good or bad thing that this works.

So I checked `elisp` dir in `emacs` itself. of 8210 `defcustoms`, 6897 have `group` in their definition with the simple regex search `\(defcustom[\s\S]+?:group`.

Soooooo OK let's add `:group` 😀
